core/src/test/java/com/google/googlejavaformat/java/DiagnosticTest.java

```diff
@@ -37,6 +40,11 @@
 public class DiagnosticTest {
   @Rule public TemporaryFolder testFolder = new TemporaryFolder();
 
+  @BeforeClass
+  public static void setUpLocale() throws Exception {
+    Locale.setDefault(Locale.US);
+  }
```

Comment:
I think it would be much better to pass `Locale.US` to the methods that require them instead of making it the global default. That way, other tests which expect a different default locale would be less likely to fail.

Reply:
The USian part is that the `google-java-format` diagnostics contain javac diagnostics for parse errors, and javac diagnostics are localized (e.g. this breaks if you set `Locale.JAPANESE`).

Maybe we should override the locale when formatting the javac diagnostics? I thought that was already happening, but apparently it doesn't work reliably: https://github.com/google/google-java-format/blob/fee1ed2685f0b4b60cafde2b5204b11ad5bc69c5/core/src/main/java/com/google/googlejavaformat/FormattingError.java#L56